### PR TITLE
Update to metrics 0.14 and switch to metrics-exporter-prometheus

### PR DESCRIPTION
**Pull request by rxdn**
No breaking changes, I updated the metrics crate to 0.14 and also swapped out metrics-observer-prometheus for metrics-exporter-prometheus.

Fixes an issue (seemily just) I was having where no metrics were returned.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -224,7 +222,7 @@ async fn handle_request(
     trace!("Response: {:?}", resp);
 
     #[cfg(feature = "expose-metrics")]
-    timing!(METRIC_KEY, start, end, "method"=>m.to_string(), "route"=>p, "status"=>resp.status().to_string());
+    histogram!(&METRIC_KEY[..], end - start, "method"=>m.to_string(), "route"=>p, "status"=>resp.status().to_string());
```
Comment (BlackHoleFox) [inline src/main.rs]: `.as_ref()` ? 
```suggestion
    histogram!(METRIC_KEY.as_ref(), end - start, "method"=>m.to_string(), "route"=>p, "status"=>resp.status().to_string());
```

File: src/main.rs [hunk 2]
```diff
@@ -24,16 +24,15 @@ use twilight_http::{
...
 use lazy_static::lazy_static;
 #[cfg(feature = "expose-metrics")]
-use metrics::timing;
+use metrics::histogram;
 #[cfg(feature = "expose-metrics")]
-use metrics_core::{Builder, Drain};
-#[cfg(feature = "expose-metrics")]
-use metrics_runtime::{observers::PrometheusBuilder, Receiver};
+use metrics_exporter_prometheus::{PrometheusBuilder, PrometheusHandle};
+use std::sync::Arc;
```
Comment (BlackHoleFox) [inline src/main.rs]: Probably should be under the `expose-metrics` feature flag

File: src/main.rs (hunk 1 above)
Comment (rxdn) [reply src/main.rs]: This results in the following error, any pointers?

```error[E0283]: type annotations needed
   --> src/main.rs:224:27
    |
224 |     histogram!(METRIC_KEY.as_ref(), end - start, "method"=>m.to_string(), "route"=>p, "status"=>resp.status().to_string());
    |                -----------^^^^^^--
    |                |          |
    |                |          cannot infer type for type parameter `T` declared on the trait `AsRef`
    |                this method call resolves to `&T`
    |
    = note: cannot satisfy `String: AsRef<_>`

error: aborting due to previous error
```

Comment (BlackHoleFox) [reply src/main.rs]: Ah, guess the `histogram!` macro isn't picky enough about types. I tried `METRIC_KEY.as_str()` locally and that worked.

Comment (rxdn) [reply src/main.rs]: Thanks, added to the PR

**Review by AEnterprise — APPROVED** [review]

**Pull request merged**